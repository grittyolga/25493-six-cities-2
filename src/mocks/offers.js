
const Offers = [
  {
    mark: `Premium`,
    image: `img/apartment-01.jpg`,
    price: 120,
    bookmark: false,
    rating: 93,
    name: `Beautiful & luxurious apartment at great location`,
    type: `Apartment`,
    ratingNumber: 4.8,
    placeType: `Entire place`,
    bedrooms: 3,
    maxAdults: 4,
    inside: [
      `Wi-Fi`,
      `Washing machine`,
      `Towels`,
      `Heating`,
      `Coffee machine`,
      `Baby seat`,
      `Kitchen`,
      `Dishwasher`,
      `Cabel TV`,
      `Fridge`
    ],
    hostName: `Angelina`,
    hostLevel: `Pro`,
    description: [`
                A quiet cozy and picturesque that hides behind a a river by the
                unique lightness of Amsterdam. The building is green and from
                18th century`, `An independent House, strategically located between Rembrand
                Square and National Opera, but where the bustle of the city
                comes to rest in this alley flowery and colorful.
             `]
  },
  {
    image: `img/room.jpg`,
    price: 80,
    bookmark: true,
    rating: 80,
    name: `Wood and stone place`,
    type: `Private room`,
    ratingNumber: 4,
    placeType: `Entire place`,
    bedrooms: 3,
    maxAdults: 4,
    inside: [
      `Wi-Fi`,
      `Washing machine`,
      `Towels`,
      `Heating`,
      `Coffee machine`,
      `Baby seat`,
      `Kitchen`,
      `Dishwasher`,
      `Cabel TV`,
      `Fridge`
    ],
    hostName: `Angelina`,
    hostLevel: `Pro`,
    description: [`
                A quiet cozy and picturesque that hides behind a a river by the
                unique lightness of Amsterdam. The building is green and from
                18th century`, `An independent House, strategically located between Rembrand
                Square and National Opera, but where the bustle of the city
                comes to rest in this alley flowery and colorful.
             `]
  },
  {
    image: `img/apartment-02.jpg`,
    price: 132,
    bookmark: false,
    rating: 80,
    name: `Canal View Prinsengracht`,
    type: `Apartment`,
    ratingNumber: 4,
    placeType: `Entire place`,
    bedrooms: 3,
    maxAdults: 4,
    inside: [
      `Wi-Fi`,
      `Washing machine`,
      `Towels`,
      `Heating`,
      `Coffee machine`,
      `Baby seat`,
      `Kitchen`,
      `Dishwasher`,
      `Cabel TV`,
      `Fridge`
    ],
    hostName: `Angelina`,
    hostLevel: `Pro`,
    description: [`
                A quiet cozy and picturesque that hides behind a a river by the
                unique lightness of Amsterdam. The building is green and from
                18th century`, `An independent House, strategically located between Rembrand
                Square and National Opera, but where the bustle of the city
                comes to rest in this alley flowery and colorful.
             `]
  },
  {
    mark: `Premium`,
    image: `img/apartment-01.jpg`,
    price: 180,
    bookmark: false,
    rating: 100,
    name: `Nice, cozy, warm big bed apartment`,
    type: `Apartment`,
    ratingNumber: 5,
    placeType: `Entire place`,
    bedrooms: 3,
    maxAdults: 4,
    inside: [
      `Wi-Fi`,
      `Washing machine`,
      `Towels`,
      `Heating`,
      `Coffee machine`,
      `Baby seat`,
      `Kitchen`,
      `Dishwasher`,
      `Cabel TV`,
      `Fridge`
    ],
    hostName: `Angelina`,
    hostLevel: `Pro`,
    description: [`
                A quiet cozy and picturesque that hides behind a a river by the
                unique lightness of Amsterdam. The building is green and from
                18th century`, `An independent House, strategically located between Rembrand
                Square and National Opera, but where the bustle of the city
                comes to rest in this alley flowery and colorful.
             `]
  },
];


export default Offers;
