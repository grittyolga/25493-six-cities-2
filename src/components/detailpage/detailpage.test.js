import React from "react";
import renderer from "react-test-renderer";
import DetailPage from "../detailpage/detailpage.jsx";
import offers from "../../mocks/offers.js";

it(`DetailPage correctly renders after relaunch`, () => {
  const tree = renderer
    .create(<DetailPage
      card={offers[0]}
    />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
