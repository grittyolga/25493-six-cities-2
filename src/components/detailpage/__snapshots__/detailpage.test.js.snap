// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailPage correctly renders after relaunch 1`] = `
<div
  className="page"
>
  <header
    className="header"
  >
    <div
      className="container"
    >
      <div
        className="header__wrapper"
      >
        <div
          className="header__left"
        >
          <a
            className="header__logo-link"
            href="main.html"
          >
            <img
              alt="6 cities logo"
              className="header__logo"
              height="41"
              src="img/logo.svg"
              width="81"
            />
          </a>
        </div>
        <nav
          className="header__nav"
        >
          <ul
            className="header__nav-list"
          >
            <li
              className="header__nav-item user"
            >
              <a
                className="header__nav-link header__nav-link--profile"
                href="#"
              >
                <div
                  className="header__avatar-wrapper user__avatar-wrapper"
                />
                <span
                  className="header__user-name user__name"
                >
                  Oliver.conner@gmail.com
                </span>
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </header>
  <main
    className="page__main page__main--property"
  >
    <section
      className="property"
    >
      <div
        className="property__gallery-container container"
      >
        <div
          className="property__gallery"
        >
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="img/room.jpg"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="img/apartment-01.jpg"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="img/apartment-02.jpg"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="img/apartment-03.jpg"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="img/studio-01.jpg"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="img/apartment-01.jpg"
            />
          </div>
        </div>
      </div>
    </section>
  </main>
  <div
    className="property__container container"
  >
    <div
      className="property__wrapper"
    >
      <div
        className="property__mark"
      >
        <span>
          Premium
        </span>
      </div>
      <div
        className="property__name-wrapper"
      >
        <h1
          className="property__name"
        >
          Beautiful & luxurious apartment at great location
        </h1>
        <button
          className="property__bookmark-button button"
          type="button"
        >
          <svg
            className="property__bookmark-icon"
            height="33"
            width="31"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="property__rating rating"
      >
        <div
          className="property__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "96%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
        <span
          className="property__rating-value rating__value"
        >
          4.8
        </span>
      </div>
      <ul
        className="property__features"
      >
        <li
          className="property__feature property__feature--entire"
        >
          Entire place
        </li>
        <li
          className="property__feature property__feature--bedrooms"
        >
          3
           Bedrooms
        </li>
        <li
          className="property__feature property__feature--adults"
        >
          Max 
          4
           adults
        </li>
      </ul>
      <div
        className="property__price"
      >
        <b
          className="property__price-value"
        >
          €
          120
          0
        </b>
        <span
          className="property__price-text"
        >
           night
        </span>
      </div>
      <div
        className="property__inside"
      >
        <h2
          className="property__inside-title"
        >
          What's inside
        </h2>
        <ul
          className="property__inside-list"
        >
          <li
            className="property__inside-item"
          >
            Wi-Fi
          </li>
          <li
            className="property__inside-item"
          >
            Washing machine
          </li>
          <li
            className="property__inside-item"
          >
            Towels
          </li>
          <li
            className="property__inside-item"
          >
            Heating
          </li>
          <li
            className="property__inside-item"
          >
            Coffee machine
          </li>
          <li
            className="property__inside-item"
          >
            Baby seat
          </li>
          <li
            className="property__inside-item"
          >
            Kitchen
          </li>
          <li
            className="property__inside-item"
          >
            Dishwasher
          </li>
          <li
            className="property__inside-item"
          >
            Cabel TV
          </li>
          <li
            className="property__inside-item"
          >
            Fridge
          </li>
        </ul>
      </div>
      <div
        className="property__host"
      >
        <h2
          className="property__host-title"
        >
          Meet the host
        </h2>
        <div
          className="property__host-user user"
        >
          <div
            className="property__avatar-wrapper property__avatar-wrapper--pro user__avatar-wrapper"
          >
            <img
              alt="Host avatar"
              className="property__avatar user__avatar"
              height="74"
              src="img/avatar-angelina.jpg"
              width="74"
            />
          </div>
          <span
            className="property__user-name"
          >
            Angelina
          </span>
          <span
            className="property__user-status"
          >
            Pro
          </span>
        </div>
        <div
          className="property__description"
        >
          <p
            className="property__text"
          >
            
                A quiet cozy and picturesque that hides behind a a river by the
                unique lightness of Amsterdam. The building is green and from
                18th century
          </p>
          <p
            className="property__text"
          >
            An independent House, strategically located between Rembrand
                Square and National Opera, but where the bustle of the city
                comes to rest in this alley flowery and colorful.
             
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
